content/learn/book/your-first-game/_index.md

```diff
@@ -0,0 +1,688 @@
+            sprite: Sprite::new(Vec2::new(120.0, 30.0)),
+            ..Default::default()
+        })
+        .with(Paddle { speed: 500.0 })
+        .with(Collider::Paddle);
+}
+
+fn paddle_movement_system(
+    time: Res<Time>,
+    keyboard_input: Res<Input<KeyCode>>,
+    mut query: Query<(&Paddle, &mut Transform)>,
+) {
+    for (paddle, mut transform) in query.iter_mut() {
+        let mut direction = 0.0;
+        if keyboard_input.pressed(KeyCode::Left) {
+            direction -= 1.0;
+        }
+
+        if keyboard_input.pressed(KeyCode::Right) {
+            direction += 1.0;
+        }
+
+        let translation = &mut transform.translation;
+        // move the paddle horizontally
+        translation.x += time.delta_seconds() * direction * paddle.speed;
+        // bound the paddle
+        translation.x = translation.x.min(380.0).max(-380.0);
+    }
+}
+
+```
+
+The code might be a lot to take in at first glance, but don't worry. We will take it one section at a time.
+
+Before we do that however, let's see it in action!
+
+Create a new folder and project on your hard drive using this command:
+
+```bash
+cargo new my_breakout_game
+cd my_breakout_game
+```
+Edit Cargo.toml to add a dependency on the Bevy git repo as outline in the [getting started guide](/learn/book/getting-started/).
+
+```toml
+[dependencies]
+bevy={git="https://github.com/bevyengine/bevy", rev="bc4fe9b186db3f32eef38fc4241289df480fa949"}
+```
+(We do this because for the code above Bevy 0.4 is already outdated.)
+
+Then copy and paste the large piece of code above into the main.rs file in the my_breakout_game/src folder.
+
+Finally execute this in your project folder:
+
+```bash
+cargo run
+```
+You will probably get 3 warnings. They are okay for now, we will have them fixed by the end of this tutorial.  
+
+You should be able to move the paddle with your cursor keys and it should look like this:
+
+![screenshot2](breakout2.png)
+
+Okay, so what is the code doing?
+
+```rs
+use bevy::{
+    prelude::*,
+    render::pass::ClearColor,
+};
+```
+This simply declares a couple of local name bindings to the most common Bevy types and plugins as well as a local name binding to ClearColor.
+ClearColor is a struct that contains the color we will use to clear the game with; effectively becoming the background color of our game.
+
+```rs
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .insert_resource(ClearColor(Color::rgb(0.9, 0.9, 0.9)))
+        .add_startup_system(setup.system())
+        .add_system(paddle_movement_system.system())
+        .run();
+}
+```
+This is the main entry point into our program. This function will be called once the program is successfully compiled and executed.
+
+[App::build()](https://github.com/bevyengine/bevy/blob/main/crates/bevy_app/src/app.rs) is a very simple function that just returns a new instance of [AppBuilder](https://docs.rs/bevy/0.4.0/bevy/app/struct.AppBuilder.html)
```

Comment:
```suggestion
[App::build()](https://docs.rs/bevy/latest/bevy/app/struct.App.html) is a very simple function that just returns a new instance of [AppBuilder](https://docs.rs/bevy/latest/bevy/app/struct.AppBuilder.html)
```
I think links like this should always go to docs.rs.
Also using latest here, makes it unnecesary to update the link on every update.

Reply:
In this case I deliberately chose to show the actual source code of the function.
I  think it is very educational to sometimes see the source code and not just the parameter documentation.
This is one of those cases in my opinion.

Reply:
I would probably include both links here. Be sure to link to `latest` rather than `main` on GitHub though!